### PR TITLE
chore: CRP-2614 Rename IDKG test functions and struct fields in Network and SubnetTopology

### PR DESCRIPTION
**Pull request by eichhorl**
This PR updates some functions and types from "IDKG" naming to a more general "Chain Key" naming, which is also applicable for VetKd. As only the names are altered, the protobuf changes are backwards/forwards compatible. Therefore it is safe to use `CI_OVERRIDE_BUF_BREAKING` in this PR.

### CODE REVIEW

**Review by mraszyk — APPROVED** [review]

**Review by rumenov — APPROVED** [review]

**Review by dsarlis — APPROVED** [review]

**Review by schneiderstefan — APPROVED** [review]

File: rs/protobuf/def/state/metadata/v1/metadata.proto [hunk 1]
```diff
@@ -50,7 +50,7 @@ message NetworkTopology {
   reserved "ecdsa_signing_subnets";
   repeated types.v1.CanisterId bitcoin_testnet_canister_ids = 6;
   repeated types.v1.CanisterId bitcoin_mainnet_canister_ids = 7;
-  repeated IDkgKeyEntry idkg_signing_subnets = 8;
+  repeated ChainKeySubnetEntry chain_key_signing_subnets = 8;
```
[the remark this answers is not in the document]
Comment (eichhorl) [reply rs/protobuf/def/state/metadata/v1/metadata.proto]: Good point, done

File: rs/replicated_state/src/metadata_state.rs [hunk 2]
```diff
@@ -191,9 +191,9 @@ pub struct NetworkTopology {
     pub canister_migrations: Arc<CanisterMigrations>,
     pub nns_subnet_id: SubnetId,
 
-    /// Mapping from iDKG key_id to a list of subnets which can sign with the
+    /// Mapping from master public key_id to a list of subnets which can sign with the
```
Comment (fspreiss) [inline rs/replicated_state/src/metadata_state.rs:194]: nit:
* "sign with" -> "use" (or similar)
* "signing subnets" -> "chain-key enabled subnets" (or similar)

File: rs/state_machine_tests/src/lib.rs [hunk 3]
```diff
@@ -279,7 +279,7 @@ pub fn finalize_registry(
 fn make_nodes_registry(
     subnet_id: SubnetId,
     subnet_type: SubnetType,
-    idkg_keys_signing_enabled_status: &BTreeMap<MasterPublicKeyId, bool>,
+    chain_keys_signing_enabled_status: &BTreeMap<MasterPublicKeyId, bool>,
```
Comment (fspreiss) [inline rs/state_machine_tests/src/lib.rs:282]: nit: How about chain_keys_signing_enabled_status -> chain_keys_enabled_status ?

**Review by fspreiss — APPROVED** [review]
Thanks, @eichhorl!

Maybe add a brief explanation in the PR description why it is safe to use `CI_OVERRIDE_BUF_BREAKING`.

File: rs/replicated_state/src/metadata_state.rs (hunk 2 above)
Comment (eichhorl) [reply rs/replicated_state/src/metadata_state.rs]: Thanks!

File: rs/state_machine_tests/src/lib.rs (hunk 3 above)
Comment (eichhorl) [reply rs/state_machine_tests/src/lib.rs]: I missed those, thanks a lot

**Review by Sawchord — APPROVED** [review]

**Pull request merged**